### PR TITLE
Clarify product tier requirement

### LINKED ISSUE #14851: Clarify product tier requirement in 'Managing code review settings for your team > About Auto Assignment'
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#about-auto-assignment

### What part(s) of the article would you like to see updated?

**About Auto Assignment**

> To further enhance your team's collaboration abilities, you can upgrade to GitHub Enterprise Cloud, which includes features like protected branches and code owners on private repositories. For more information about how you can try GitHub Enterprise Cloud for free, see "Setting up a trial of GitHub Enterprise Cloud."

Protected branches and code owners on private repositories are no longer features of just GHEC. Caused some confusion when trying to understand whether or not I could use these in my Github Team account.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by asmitbm**
### Why:
Clarified product tier requirement in 'Managing code review settings for your team > About Auto Assignment'.
There's no difference between Team and Cloud on those collaboration features, as you can see at https://github.com/pricing.

Closes: #14851 

### What's being changed:

A para from the About auto assignment have been removed, as collaboration features mentioned there was part of both GitHub Team and GitHub Enterprise Cloud account.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@Asmit2952 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md
Comment (prensjr) [inline content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md:40]: All I good but am trying to get to know how to do this?

**Review by prensjr — COMMENTED** [review]
All good thanks

File: content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md
Comment (prensjr) [reply content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md:40]: How do I do this?

Comment (Asmit2952) [reply content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md:40]: What do you want to know?

**Comment by MAC-ARTHUR-TRUE** [comment]
Madison chambers or Chamberlin POW MIA Afghanistan IRAC,  impersonated by Pamela mceandry, calling themselves Steve and Mary Female identity change to Paris Pichai or Pichilli Apaguasa. Secured identity with "memory" software accounts of DreamWorks high jacked used against her by known criminals and theives.Stalking of female documented with federal restraining order permanent. Paris 304 663 4110 Sent from my  U.S.Cellular© Smartphone-------- Original message --------From: mc4928184 ***@***.***> Date: 2/28/22  15:53  (GMT-05:00) To: ***@***.*** Subject: FW: Urgent HelpSent from my  U.S.Cellular© Smartphone-------- Original message --------From: mc4928184 ***@***.***> Date: 2/28/22  3:51 PM  (GMT-05:00) To: ***@***.*** Subject: FW: Urgent Data pushesMasked ipPhone info changesAll obsevedSent from my  U.S.Cellular© Smartphone-------- Original message --------From: mc4928184 ***@***.***> Date: 2/24/22  12:07 AM  (GMT-05:00) To: ***@***.*** Subject: Urgent Crime Sent from my  U.S.Cellular© SmartphoneSent from my  U.S.Cellular© Smartphone
-------- Original message --------From: Edward Prens Jr ***@***.***> Date: 3/3/22  15:53  (GMT-05:00) To: github/docs ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [github/docs] Clarify product tier requirement (PR #15991) 
@prensjr commented on this pull request.



In content/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team.md:
> @@ -36,10 +36,6 @@ When you enable auto assignment, any time your team has been requested to review
 
 When code owners are automatically requested for review, the team is still removed and replaced with individuals unless a branch protection rule is configured to require review from code owners. If such a branch protection rule is in place, the team request cannot be removed and so the individual request will appear in addition.
 
-{% ifversion fpt %}
-To further enhance your team's collaboration abilities, you can upgrade to {% data variables.product.prodname_ghe_cloud %}, which includes features like protected branches and code owners on private repositories. {% data reusables.enterprise.link-to-ghec-trial %}
-{% endif %}
-
 ### Routing algorithms
 

How do I do this?

**Review by felicitymay — APPROVED** [review]
@Asmit2952 - thank you for updating this article to reflect a change to the product tiers 💖 

This looks great and we'll get this merged.

**Pull request merged**
